ai-assistant-settings/scripts/sync-settings.mts

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env ts-node
+
+/**
+ * Updates the local settings files with values from the AI Assistants' settings at platform.openai.com.
+ * 
+ * Usage:
+ *   node --loader ts-node/esm sync-settings.mts
+ * 
+ * TODO: Make it so `--loader ts-node/esm` isn't required for this to run since Custom ESM Loaders is an
+ * experimental feature and could change.
+ */
+
+import { OpenAI } from "openai";
+import fs from "fs";
+import process from "node:process";
+import path from "path";
+import dotenv from "dotenv";
+
+dotenv.config({ path: "../../.env" });
+
+const requiredEnvVars = ["REACT_APP_OPENAI_API_KEY", "REACT_APP_OPENAI_BASE_URL"];
+const missingEnvVars = requiredEnvVars.filter((key) => !process.env[key]);
+
+if (missingEnvVars.length > 0) {
+  console.error("Missing required environment variables:");
+  missingEnvVars.forEach((key) => console.error(`- ${key}`));
+  process.exit(1);
+}
+
+const openai = new OpenAI({
+  apiKey: process.env.REACT_APP_OPENAI_API_KEY || "fake-key",
+  baseURL: process.env.REACT_APP_OPENAI_BASE_URL,
+  organization: "org-jbU1egKECzYlQI73HMMi7EOZ",
+  project: "proj_VsykADfoZHvqcOJUHyVAYoDG",
+});
+
+const getAssistants = async () => {
+  try {
+    const { data } = await openai.beta.assistants.list();
+    return data;
+  } catch (err) {
+    console.error("Error fetching assistants:", err);
+    return [];
+  }
+};
+
+const writeAssistantFiles = (assistant: any) => {
+  const assistantDir = path.resolve("../assistants", assistant.id);
```

Comment:
I'm not sure, but does mean it is sensitive to the working directory at the time you run the script? If so, you should probably document what's expected.

Reply:
@bgoldowsky Yeah, that is true. I've updated the script so it will work no matter what the working directory is. That allowed me to also set up an npm script in package.json for it. So now it can be run with `npm run sync:assistant-settings`.

Reply:
Nice! 👍